File: server/src/test/java/com/pass/unit/exam/domain/ExamDescriptionTest.java
package com.pass.unit.exam.domain;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import com.pass.exam.domain.ExamDescription;
import com.pass.global.exception.BadRequestException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class ExamDescriptionTest {

    @Test
    @DisplayName("시험 설명은 생성할 수 있다.")
    void create() {
        // given
        String description = "a".repeat(2000);

        // when
        ExamDescription examDescription = new ExamDescription(description);

        // then
        assertThat(examDescription.getValue()).isEqualTo(description);
    }

    @Test
    @DisplayName("시험 설명은 null일 수 없다.")
    void descriptionNotNull() {
        assertThatThrownBy(() -> new ExamDescription(null))
                .isInstanceOf(BadRequestException.class)
                .hasMessage("시험 설명은 필수입니다.");
    }

    @Test
    @DisplayName("시험 설명은 2000자 이하여야 한다.")
    void descriptionLength() {
        // given
        String description = "a".repeat(2001);

        // when & then
        assertThatThrownBy(() -> new ExamDescription(description))
                .isInstanceOf(BadRequestException.class)
                .hasMessage("시험 설명은 2000자 이하여야 합니다.");
    }
}

File: server/src/test/kotlin/com/fluffy/exam/domain/ExamDescriptionTest.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.BadRequestException
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class ExamDescriptionTest : StringSpec({

    "시험 설명이 정상적으로 생성된다." {
        // given
        val description = "a".repeat(2000)

        // when
        val examDescription = ExamDescription(description)

        // then
        examDescription.value shouldBe description
    }

    "시험 설명은 비어있을 수 있다." {
        // given
        val description = ""

        // when
        val examDescription = ExamDescription(description)

        // then
        examDescription.value shouldBe description
    }

    "시험 설명은 2000자 이하여야 한다." {
        // given
        val description = "a".repeat(2001)

        // when & then
        shouldThrow<BadRequestException> { ExamDescription(description) }
            .message shouldBe "시험 설명은 2000자 이하여야 합니다."
    }
})